Smapp does not validate coinbase address and Node fails to run

### Environment
* Smapp version: 0.2.7
* Node version: v0.2.18-beta.0

### Describe the bug
When the User previously set up smeshing on the previous version of Smapp, he has an old coinbase address type in the `node-config.json`. Meanwhile, the new version of Smapp just copies this coinbase address when connecting to the new network and tries to use this. The node exits with panic that the address is incorrect.

### Steps to reproduce
1.  Set up the HEX-style address (0x...) in `smesher-coinbase` instead of bech32 address in `node-config.json`
2.  Run Smapp and unlock the wallet

### Expected behavior
Smapp should convert the old-style address into a Bech32 address and use it.
Or at least clean up an invalid address and set `smeshing-start` to `false` before running the Node.

### Actual behavior
The node exits with panic.

### Logs
```
2022-11-05T23:27:32.417+0300	ERROR	1d558.app          	Fatal: goroutine panicked. Stacktrace: goroutine 1 [running]:
runtime/debug.Stack()
	/opt/hostedtoolcache/go/1.18.2/x64/src/runtime/debug/stack.go:24 +0x65
github.com/spacemeshos/go-spacemesh/log.Log.Panic({0xc000128e00?, {0x15e4802?, 0x0?}}, {0x1795e56, 0x30}, {0xc000fe9240, 0x2, 0x2})
	/home/runner/work/go-spacemesh/go-spacemesh/log/zap.go:46 +0xd7
github.com/spacemeshos/go-spacemesh/cmd/node.(*App).initServices(0xc0000cc8c0, {0x1a68070, 0xc0008ca280}, {0x1d, 0x55, 0x85, 0xe1, 0x7a, 0xf9, 0x3e, ...}, ...)
	/home/runner/work/go-spacemesh/go-spacemesh/cmd/node/node.go:609 +0x2ca5
github.com/spacemeshos/go-spacemesh/cmd/node.(*App).Start(0xc0000cc8c0)
	/home/runner/work/go-spacemesh/go-spacemesh/cmd/node/node.go:1060 +0x1105
github.com/spacemeshos/go-spacemesh/cmd/node.glob..func1.1()
	/home/runner/work/go-spacemesh/go-spacemesh/cmd/node/node.go:121 +0x73
github.com/spacemeshos/go-spacemesh/cmd/node.glob..func1(0x23a2840?, {0x15e692c?, 0x5?, 0x5?})
	/home/runner/work/go-spacemesh/go-spacemesh/cmd/node/node.go:127 +0x2bc
github.com/spf13/cobra.(*Command).execute(0x23a2840, {0xc00003c1f0, 0x5, 0x5})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:876 +0x67b
github.com/spf13/cobra.(*Command).ExecuteC(0x23a2840)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
	/home/runner/work/go-spacemesh/go-spacemesh/main.go:24 +0xbd
	{"node_id": "1d5585e17af93e2798867b3a50177983df021a04f07569ca2a431e68dcaeea7b", "module": "app"}
2022-11-05T23:27:32.418+0300	PANIC	1d558.app          	failed to parse CoinbaseAccount address `0xe04c3f7e1b6229ce7ff4fc5eaa87aeabd74dcad5`: error decoding bech32: invalid character not part of charset: 98	{"node_id": "1d5585e17af93e2798867b3a50177983df021a04f07569ca2a431e68dcaeea7b", "module": "app"}
panic: failed to parse CoinbaseAccount address `0xe04c3f7e1b6229ce7ff4fc5eaa87aeabd74dcad5`: error decoding bech32: invalid character not part of charset: 98

goroutine 1 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x0?, 0x0?, {0x0?, 0x1?, 0x0?})
	/home/runner/go/pkg/mod/go.uber.org/zap@v1.23.0/zapcore/entry.go:198 +0x65
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0010721a0, {0x0, 0x0, 0x0})
	/home/runner/go/pkg/mod/go.uber.org/zap@v1.23.0/zapcore/entry.go:264 +0x3ec
go.uber.org/zap.(*SugaredLogger).log(0xc000fe8760, 0x4, {0x1795e56?, 0x0?}, {0xc000fe9240?, 0x2?, 0x2?}, {0x0, 0x0, 0x0})
	/home/runner/go/pkg/mod/go.uber.org/zap@v1.23.0/sugar.go:287 +0xee
go.uber.org/zap.(*SugaredLogger).Panicf(...)
	/home/runner/go/pkg/mod/go.uber.org/zap@v1.23.0/sugar.go:181
github.com/spacemeshos/go-spacemesh/log.Log.Panic({0xc000128e00?, {0x15e4802?, 0x0?}}, {0x1795e56, 0x30}, {0xc000fe9240, 0x2, 0x2})
	/home/runner/work/go-spacemesh/go-spacemesh/log/zap.go:47 +0x1fa
github.com/spacemeshos/go-spacemesh/cmd/node.(*App).initServices(0xc0000cc8c0, {0x1a68070, 0xc0008ca280}, {0x1d, 0x55, 0x85, 0xe1, 0x7a, 0xf9, 0x3e, ...}, ...)
	/home/runner/work/go-spacemesh/go-spacemesh/cmd/node/node.go:609 +0x2ca5
github.com/spacemeshos/go-spacemesh/cmd/node.(*App).Start(0xc0000cc8c0)
	/home/runner/work/go-spacemesh/go-spacemesh/cmd/node/node.go:1060 +0x1105
github.com/spacemeshos/go-spacemesh/cmd/node.glob..func1.1()
	/home/runner/work/go-spacemesh/go-spacemesh/cmd/node/node.go:121 +0x73
github.com/spacemeshos/go-spacemesh/cmd/node.glob..func1(0x23a2840?, {0x15e692c?, 0x5?, 0x5?})
	/home/runner/work/go-spacemesh/go-spacemesh/cmd/node/node.go:127 +0x2bc
github.com/spf13/cobra.(*Command).execute(0x23a2840, {0xc00003c1f0, 0x5, 0x5})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:876 +0x67b
github.com/spf13/cobra.(*Command).ExecuteC(0x23a2840)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
	/home/runner/work/go-spacemesh/go-spacemesh/main.go:24 +0xbd
```

--- a/desktop/main/NodeConfig.ts
+++ b/desktop/main/NodeConfig.ts
@@ -4,6 +4,7 @@ import { NodeConfig } from '../../shared/types';
 import { configCodecByPath } from '../../shared/utils';
 import { fetchNodeConfig } from '../utils';
 import { NODE_CONFIG_FILE } from './constants';
+import { safeSmeshingOpts } from './smeshingOpts';
 
 export const loadNodeConfig = async () =>
   existsSync(NODE_CONFIG_FILE)
@@ -14,7 +15,10 @@ export const loadNodeConfig = async () =>
         )
     : {};
 
-const loadSmeshingOpts = async () => (await loadNodeConfig()).smeshing;
+const loadSmeshingOpts = async () => {
+  const opts = (await loadNodeConfig()).smeshing;
+  return safeSmeshingOpts(opts);
+};
 
 export const downloadNodeConfig = async (networkConfigUrl: string) => {
   const nodeConfig = await fetchNodeConfig(networkConfigUrl);

--- a/desktop/main/smeshingOpts.ts
+++ b/desktop/main/smeshingOpts.ts
@@ -1,0 +1,47 @@
+import Bech32 from '@spacemesh/address-wasm';
+import { fromHexString } from '../../shared/utils';
+
+export type SmeshingOpts = {
+  'smeshing-coinbase': string;
+  'smeshing-opts': {
+    'smeshing-opts-datadir': string;
+    'smeshing-opts-numfiles': number;
+    'smeshing-opts-numunits': number;
+    'smeshing-opts-provider': number;
+    'smeshing-opts-throttle': boolean;
+  };
+  'smeshing-start': boolean;
+};
+
+export const isSmeshingOpts = (a: any): a is SmeshingOpts =>
+  a &&
+  typeof a['smeshing-coinbase'] === 'string' &&
+  typeof a['smeshing-start'] === 'boolean' &&
+  typeof a['smeshing-opts'] === 'object' &&
+  typeof a['smeshing-opts']['smeshing-opts-datadir'] === 'string' &&
+  typeof a['smeshing-opts']['smeshing-opts-numfiles'] === 'number' &&
+  typeof a['smeshing-opts']['smeshing-opts-numunits'] === 'number' &&
+  typeof a['smeshing-opts']['smeshing-opts-provider'] === 'number' &&
+  typeof a['smeshing-opts']['smeshing-opts-throttle'] === 'boolean' &&
+  a['smeshing-coinbase'].length > 0 &&
+  a['smeshing-opts']['smeshing-opts-numfiles'] >= 1 &&
+  a['smeshing-opts']['smeshing-opts-numunits'] >= 1 &&
+  a['smeshing-opts']['smeshing-opts-provider'] >= 0;
+
+export const safeSmeshingOpts = (opts: SmeshingOpts) => {
+  if (!isSmeshingOpts(opts)) return {};
+
+  const oCoinbase = opts['smeshing-coinbase'];
+  const coinbase =
+    typeof oCoinbase === 'string' && oCoinbase.startsWith('0x')
+      ? Bech32.generateAddress(fromHexString(oCoinbase))
+      : oCoinbase;
+  if (Bech32.verify(coinbase)) {
+    return {
+      ...opts,
+      'smeshing-coinbase': coinbase,
+    };
+  } else {
+    return {};
+  }
+};

